src/api/fungible_asset.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { GetFungibleAssetMetadataResponse, HexInput, PaginationArgs } from "../types";
+import { AptosConfig } from "./aptos_config";
+import { getFungibleAssetMetadata } from "../internal/fungible_asset";
+
+/**
+ * A class to query all `FungibleAsset` related queries on Aptos.
+ */
+export class FungibleAsset {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Queries the fungible asset metadata
```

Comment:
Queries the current fungible asset metadata.  Makes it clearer that the  indexer only returns (semi current state)